### PR TITLE
Angular support for banner

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 🤨 Rationale

Angular support for banner component

## 👩‍💻 Implementation

- directive for banner
- added tests
- added to example app
- updated README

## 🧪 Testing

Automated tests and tested in example app

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: angular-workspace/projects/ni/nimble-angular/src/public-api.ts [hunk 1]
```diff
@@ -97,6 +99,7 @@ export * from './testing/async-helpers';
 
 // Export items that would otherwise be exported by multiple components just from here to avoid exporting them multiple times.
 // Can be cleaned up when switching to multiple entry points, see: https://github.com/ni/nimble/issues/172
+export { BannerSeverity } from '@ni/nimble-components/dist/esm/banner/types';
```
Comment (rajsite) [inline angular-workspace/projects/ni/nimble-angular/src/public-api.ts:102]: I believe BannerSeverity is only used by the banner directive so it should be exported just from there (not the comment above stating the exports here are for those enums that would be exported from multiple components)

File: angular-workspace/projects/ni/nimble-angular/src/directives/banner/nimble-banner.directive.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public constructor(private readonly renderer: Renderer2, private readonly elementRef: ElementRef<Banner>) {}
+
+    @HostListener('toggle', ['$event'])
+    public onToggle($event: Event): void {
+        if ($event.target === this.elementRef.nativeElement) {
+            this.openChange.emit(this.open);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (rajsite) [reply angular-workspace/projects/ni/nimble-angular/src/directives/banner/nimble-banner.directive.ts:70]: Discussed offline with molly and mert, the action for this PR is to export the type for the details property of the toggle custom event (so users can choose to bind to the toggle event directly if they need to) and keep the two-way binding set-up for the open property (which uses the toggle event to emit `@Output openChange`). Molly will update #999 to capture doing the same pattern for menu-button (binding openChange to the toggle event) and capture docs on the resolved policy for Angular events (don't create `@Output` to map to events, only for properties to do two-way binding).

**Review by rajsite — APPROVED** [review]

File: angular-workspace/projects/ni/nimble-angular/src/directives/banner/nimble-banner.directive.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (m-akinc) [reply angular-workspace/projects/ni/nimble-angular/src/directives/banner/nimble-banner.directive.ts:71]: I created PR #1065 to define and export the type from `nimble-components`. After that is submitted, I will re-export the type from `nimble-angular`.

File: angular-workspace/projects/ni/nimble-angular/src/public-api.ts (hunk 1 above)
Comment (m-akinc) [reply angular-workspace/projects/ni/nimble-angular/src/public-api.ts]: Done

File: angular-workspace/projects/ni/nimble-angular/src/directives/banner/nimble-banner.directive.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (m-akinc) [reply angular-workspace/projects/ni/nimble-angular/src/directives/banner/nimble-banner.directive.ts:72]: Now re-exporting the type.

**Pull request merged**